Octokit/Http/CustomAwaiter.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Octokit
+{
+    public static class CustomAwaiter
+    {
+        public static TaskAwaiter<IReadOnlyList<T>> GetAwaiter<T>(this IDeferredRequest<T> request)
```

Comment:
Yeah, I'm not sure I can magic up a type here that's both Hot and also modifiable using WithOptions.

So I'm weighing up varying degrees of breaking changes to see how they feel.